source/state_representation/include/state_representation/State.hpp

```diff
@@ -159,6 +159,11 @@ class State : public std::enable_shared_from_this<State> {
    */
   void set_empty(bool empty = true);
 
+  /**
+   * @brief Throw an exception if the state is empty
+   */
+  void throw_if_empty() const;
```

Comment:
suggestion: This is making an assertion, so I would prefer the following name. Also, include a `@throws exceptions::EmptyStateException` tag in the description.
```suggestion
  void assert_not_empty() const;
```